File: Accesa/Accesa.SportsBuddy/Accesa.SportsBuddy.UI/src/models/Trainee.ts
import { GenderEnum, Role, Address, UserRoleEnum } from './UserInfo';

export interface Trainee {
    id?: number;
    firstName: string;
    lastName: string;
    email: string;
    password?: string;
    addressNavigation?: Address;
    birthDate: Date;
    gender?: GenderEnum;
    phoneNumber: string;
    createdAt?: Date;
    role: Role;
    score?:number;
    picture?: string;
}

export const DEFAULT_TRAINEE = {
    id: 0,
    firstName: '',
    lastName: '',
    email: '',
    addressNavigation: {
        id: 0,
        street: '',
        city: '',
        state: '',
        postalCode: '',
        country: '',
    },
    birthDate: new Date("2021-08-18T00:00:00"),
    gender: GenderEnum.MALE,
    phoneNumber: '',
    role: {
        id: UserRoleEnum.TRAINEE,
        name: ''
    }
}

//export type TraineeRegister = Omit<Trainee, 'addressNavigation'>;

File: Accesa/Accesa.SportsBuddy/Accesa.SportsBuddy/Accesa.SportsBuddy/Database/Models/Trainee.cs
using System;
using System.Collections.Generic;

#nullable disable

namespace Accesa.SportsBuddy.Database.Models
{
    public partial class Trainee
    {
        public Trainee()
        {
            Favorites = new HashSet<Favorite>();
            Reviews = new HashSet<Review>();
            TraineeTrainingPrograms = new HashSet<TraineeTrainingProgram>();
            TrainerTrainees = new HashSet<TrainerTrainee>();
        }

        public int Id { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public DateTime BirthDate { get; set; }
        public string Gender { get; set; }
        public string PhoneNumber { get; set; }
        public string Email { get; set; }
        public string Password { get; set; }
        public int Address { get; set; }
        public DateTime CreatedAt { get; set; }
        public int RoleId { get; set; }

        public virtual Address AddressNavigation { get; set; }
        public virtual Role Role { get; set; }
        public virtual ICollection<Favorite> Favorites { get; set; }
        public virtual ICollection<Review> Reviews { get; set; }
        public virtual ICollection<TraineeTrainingProgram> TraineeTrainingPrograms { get; set; }
        public virtual ICollection<TrainerTrainee> TrainerTrainees { get; set; }
    }
}
